allauth/socialaccount/providers/github/views.py

```diff
@@ -30,9 +30,11 @@ def complete_login(self, request, app, token, **kwargs):
         )
         resp.raise_for_status()
         extra_data = resp.json()
-        if app_settings.QUERY_EMAIL and not extra_data.get("email"):
+        if app_settings.QUERY_EMAIL:
             emails = self.get_emails(headers)
             if emails:
+                if email := extra_data.get("email"):
+                    emails = [e for e in emails if e["email"] == email]
```

Comment:
IMO we could remove this.

When I was experimenting locally without this filtering, allauth got a list of 4 email addresses back from GH, but seemingly only stored either (1) the one marked as primary, or (2) the one(s) matching the existing local allauth user's email address. TBH I didn't dig deeply enough through the code to know which of those resulted in what I observed.

Reply:
Yes, we should remove this. There is a method `cleanup_email_addresses()` which handles multiple email addresses. If there are issues there, than I would prefer for those to be addressed instead.